src/main/java/org/spongepowered/api/world/World.java

```diff
@@ -262,6 +262,14 @@ default LocatableBlock getLocatableBlock(int x, int y, int z) {
     WorldBorder getWorldBorder();
 
     /**
+     * Sets the world border for this world.
+     * 
+     * @param border The world border to be used.
+     * @param cause The cause of the world border's change.
+     */
+    void setWorldBorder(WorldBorder border, Cause cause);
```

Comment:
In that case, the method is rather confusing in my opinion. It implies that `getWorldBorder()` will then return the new world border object (and that modifications to the `WorldBorder` given in here also update the world's `WorldBorder`). I'd just remove it, plugins can easily modify the world border returned by `getWorldBorder()`.